test/multiply_test.cc

```diff
@@ -292,60 +368,80 @@ template <class Routine> void TestMultiply(Index A_rows, Index width, Index B_co
+  TestMultiply<AVX2_8bit>(8, 2048, 256, 19, 19, 1.8, 1.8);
+  TestMultiply<AVX2_8bit>(320, 256, 256, .1, 1, 0.1);
+  TestMultiply<AVX2_8bit>(472, 256, 256, .1, 1, 0.1);
+  TestMultiply<AVX2_8bit>(248, 256, 256, .1, 1, 0.1);
+  TestMultiply<AVX2_8bit>(200, 256, 256, .1, 1, 0.1);
+}
+
+TEST_CASE ("Multiply AVX2 16bit", "[multiply]") {
+  if (kCPU < CPU_AVX2) return;
+  TestMultiply<AVX2_16bit>(8, 256, 256, .1, 1, 0.01);
+  TestMultiply<AVX2_16bit>(8, 2048, 256, .1, 1, 0.02);
+  TestMultiply<AVX2_16bit>(320, 256, 256, .1, 1, 0.01);
+  TestMultiply<AVX2_16bit>(472, 256, 256, .1, 1, 0.01);
+  TestMultiply<AVX2_16bit>(248, 256, 256, .1, 1, 0.01);
+  TestMultiply<AVX2_16bit>(200, 256, 256, .1, 1, 0.01);
+}
 
-// Program takes no input
-int main(int argc, char ** argv) {
-    std::srand(45678);
-    using namespace intgemm;
-    if (kCPU >= CPU_SSE2) {
-      TestTranspose16();
-    }
-    if (kCPU >= CPU_SSSE3) {
-      TestTranspose8();
-    }
 #ifndef INTGEMM_NO_AVX512
-    TestPrepare<AVX512_8bit>(64, 8);
-    TestPrepare<AVX512_8bit>(256, 32);
-    TestPrepare<AVX512_16bit>(32, 8);
-    TestPrepare<AVX512_16bit>(256, 32);
-    TestSelectColumnsB<AVX512_8bit>();
-    TestSelectColumnsB<AVX512_16bit>(256, 256);
+  TEST_CASE ("Multiply AVX512 8bit", "[multiply]") {
+    if (kCPU < CPU_AVX512BW) return;
+    TestMultiply<AVX512_8bit>(8, 256, 256, .1, 1, 0.06);
+    TestMultiply<AVX512_8bit>(8, 2048, 256, 4.2, 4, 0.41, 0.37);
+    TestMultiply<AVX512_8bit>(320, 256, 256, .1, 1, 0.06);
+    TestMultiply<AVX512_8bit>(472, 256, 256, .1, 1, 0.06);
+    TestMultiply<AVX512_8bit>(248, 256, 256, .1, 1, 0.06);
+    TestMultiply<AVX512_8bit>(200, 256, 256, .1, 1, 0.06);
+  }
+
+  TEST_CASE ("Multiply AVX512 16bit", "[multiply]") {
+    if (kCPU < CPU_AVX512BW) return;
+    TestMultiply<AVX512_16bit>(8, 256, 256, .1, 1, 0.01);
+    TestMultiply<AVX512_16bit>(8, 2048, 256, .1, 1, 0.011);
+    TestMultiply<AVX512_16bit>(320, 256, 256, .1, 1, 0.01);
+    TestMultiply<AVX512_16bit>(472, 256, 256, .1, 1, 0.01);
+    TestMultiply<AVX512_16bit>(248, 256, 256, .1, 1, 0.01);
+    TestMultiply<AVX512_16bit>(200, 256, 256, .1, 1, 0.01);
+  }
 #endif
-    TestPrepare<AVX2_8bit>(64, 32);
-    TestPrepare<AVX2_16bit>(64, 32);
-    TestSelectColumnsB<AVX2_8bit>(256, 256);
-    TestSelectColumnsB<AVX2_16bit>(256, 256);
-    TestPrepare<SSSE3_8bit>(16, 8);
-    TestPrepare<SSSE3_8bit>(32, 16);
-    TestPrepare<SSSE3_8bit>(32, 32);
-    TestSelectColumnsB<SSSE3_8bit>();
-    TestSelectColumnsB<SSSE3_8bit>(256, 256);
-    TestPrepare<SSE2_16bit>(8, 8);
-    TestPrepare<SSE2_16bit>(32, 32);
-    TestSelectColumnsB<SSE2_16bit>();
-    TestSelectColumnsB<SSE2_16bit>(256, 256);
-    TestMax<__m128>();
-    TestMaxAbsolute<SSE2_MaxAbsolute>();
-/*    if (kCPU >= CPU_AVX2) {
-      TestMaxAbsolute<AVX2_MaxAbsolute>();
-    }*/
+} // namespace intgemm
+
+int main(int argc, char ** argv) {
+    std::srand(45678); //If the random seed is not crucial we can forego the main alltogether
```

Comment:
If we're concerned about random initialization being consistent on different CPUs, why not use the C++11 random stuff and seed each time the random number generator is called?